tests/unit/query_plan.cpp

```diff
@@ -1553,4 +1553,10 @@ TYPED_TEST(TestPlanner, LabelPropertyInListWhereLabelPropertyOnRight) {
   }
 }
 
+TYPED_TEST(TestPlanner, Foreach) {
+  AstStorage storage;
+  auto *i = NEXPR("i", IDENT("i"));
+  auto *query = QUERY(SINGLE_QUERY(FOREACH(i, {})));
+  CheckPlan<TypeParam>(query, storage, ExpectForeach());
+}
```

Comment:
Add a couple of more tests here, or at least one which is a bit more complex 🧪

Reply:
Although I think the tests in `gql_behave` suffice (since they test the planning implicitly) I added a few more for completeness!